deploy/eck-operator/templates/service-monitor.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if .Values.config.metrics.secureMode.enabled }}
+{{- if and .Values.config.metrics.secureMode.enabled .Values.serviceMonitor.enabled }}
```

Comment:
I am definitely 👍  with this, but is it not a breaking change? What is going to happen during an upgrade if `serviceMonitor.enabled` is not set?

Reply:
I thought this was covered by defaulting this value to `true`, so it is backwards compatible.

Reply:
Sorry, you're right, I missed your comment in the values:

```yaml
serviceMonitor:
  # allows to disable the serviceMonitor, enabled by default for backwards compatibility
  enabled: true # for backwards compatibility
```